pkg/operator/apis/monitoring/v1/http_types.go

```diff
@@ -101,11 +101,8 @@ func (s *SecretKeySelector) toPrometheusSecretRef(m PodMonitoringCRD, pool Prome
 	}
 	if m.IsNamespaceScoped() {
 		monitoringNamespace := m.GetNamespace()
-		if monitoringNamespace == "" {
-			monitoringNamespace = metav1.NamespaceDefault
-		}
-		if ns != monitoringNamespace {
-			return "", fmt.Errorf("must use namespace %q, got: %q", m.GetNamespace(), ns)
```

Comment:
I'd like to keep this error handling because it would help catch issues in incorrect unit tests or if somehow this field was set.

We can simply it. Combined with the last suggestion, we could do:

```go
if ns == "" {
	...
} else if m.IsNamespaceScoped() && ns != m.GetNamespace() {
	return "", fmt.Errorf("must use namespace %q, got: %q", m.GetNamespace(), ns)
}
```

Reply:
1b5b60e7cc23cfd66f979c91bf2a63a3bc06d6b6